packages/contractkit/src/identity/offchain-data-wrapper.ts

```diff
@@ -74,46 +89,93 @@ export default class OffchainDataWrapper {
 
   readDataFrom = makeAsyncThrowable(this.readDataFromAsResult.bind(this))
 
-  async writeDataTo(data: string, dataPath: string) {
+  async writeDataTo(
+    data: Buffer,
+    signature: string,
```

Comment:
It may be better to define the signature as a Buffer here instead of a hex string. If we want to do this, it will be much easier to do now than later.